lib/api/core/validation/index.js

```diff
@@ -1,690 +1,708 @@
-      this
-        .curateValidatorFilter(indexName, collectionName, collectionSpec.validators, dryRun)
-        .then(validationFilter => {
-          treatedSpecification.validators = validationFilter.id;
+    });
 
-          return resolve(treatedSpecification);
-        })
-        .catch(error => {
-          this.kuzzle.pluginsManager.trigger('log:error', error);
-          return reject(new InternalError(`Validator specification of the collection ${indexName}.${collectionName} triggered an error`));
-        });
+    if (errors.length > 0) {
+      return {isValid: false, errors: errors};
     }
-    else {
-      return resolve(treatedSpecification);
-    }
-  });
-};
 
-Validation.prototype.structureCollectionValidation = function validationStructureCollectionValidation (collectionSpec, indexName, collectionName, verboseErrors) {
-  var
-    result,
-    errors = [],
-    fields = {},
-    maxDepth = 0;
-
-  verboseErrors = verboseErrors || false;
-
-  Object.keys(collectionSpec.fields).forEach(fieldName => {
-    var
-      field,
-      // We deep clone the field because we will modify it
-      fieldSpecClone = _.cloneDeep(collectionSpec.fields[fieldName]);
-
-    try {
-      result = this.curateFieldSpecification(fieldSpecClone, indexName, collectionName, fieldName, verboseErrors);
-      if (result.isValid === false) {
-        errors = _.concat(errors, result.errors);
-        this.kuzzle.pluginsManager.trigger('log:error', result.errors.join('\n'));
-      } else {
-        field = result.fieldSpec;
-        field.path = fieldName.split('/');
-        field.depth = field.path.length;
-        if (field.depth > maxDepth) {
-          maxDepth = field.depth;
-        }
-
-        if (!fields[field.depth]) {
-          fields[field.depth] = [];
-        }
-
-        fields[field.depth].push(field);
-      }
-    }
-    catch (error) {
-      this.kuzzle.pluginsManager.trigger('log:error', error);
-      throwOrStoreError(`Specification for the field ${indexName}.${collectionName}.${fieldName} triggered an error`, verboseErrors, errors);
+    if (Object.keys(fields).length > 0) {
+      return curateStructuredFields(this.typeAllowsChildren, fields, maxDepth);
     }
-  });
 
-  if (errors.length > 0) {
-    return {isValid: false, errors: errors};
+    return {};
   }
 
-  if (Object.keys(fields).length > 0) {
-    return curateStructuredFields(this.typeAllowsChildren, fields, maxDepth);
-  }
+  /**
+   * @param {FieldSpecification} fieldSpec
+   * @param {string} indexName
+   * @param {string} collectionName
+   * @param {string} fieldName
+   * @param {boolean} verboseErrors
+   * @returns {object}
+   * @throws InternalError
+   */
+  curateFieldSpecification(fieldSpec, indexName, collectionName, fieldName, verboseErrors) {
+    var
+      returnedTypeOptions,
+      result,
+      errors = [];
 
-  return {};
-};
+    verboseErrors = verboseErrors || false;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "verboseErrors". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1226)